.github/workflows/test.yml

```diff
@@ -7,8 +7,18 @@ env:
 
 jobs:
   test:
-    name: Test
-    runs-on: ubuntu-latest
+    name: Test on ${{ matrix.os }} (using rustc ${{ matrix.rust }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        rust:
+          - 1.53.0
+          - stable
+          - nightly
```

Comment:
GitHub won't let me comment on an unchanged line but I think you missed replacing `toolchain: stable` with `toolchain: ${{matrix.rust }}` on line 29.

Also the cache key on line 47 should be updated with the rust version because different versions of rust are incompatible.
Something like:
```yaml
key: ${{ runner.os }}-cargo-${{ matrix.rust }}-${{ hashFiles('**/Cargo.lock') }}
```